datafusion/core/src/datasource/listing/url.rs

```diff
@@ -246,6 +248,9 @@ mod tests {
         let url = ListingTableUrl::parse("file:///foo").unwrap();
         let child = Path::parse("/foob/bar").unwrap();
         assert!(url.strip_prefix(&child).is_none());
+
+        let url = ListingTableUrl::parse("file:///with space/foo/bar").unwrap();
```

Comment:
Could we get a test of a non-ASCII character such as a percent encoded emoji or something

We should also test non-URL safe characters like a percent encoded `?`

Reply:
Thanks. I added some test cases at 439f6dc